### PR TITLE
Feature: managed aliases

### PR DESCRIPTION
**Pull request by Davidmattei**
Environment local and external should only use **1** index, but we want to manage aliases with multiple indexes (indices).

On the environment page we can now view/create/edit and delete managed aliases. Managed aliases are stored in 'managed_alias' table. It's also possible to align managed aliases on their edit page.

Added indexes counter to the unreferenced aliases overview. Disabled the attach function for unreferenced aliases with multiple indexes. And also added the delete function for unreferenced aliases.

### CODE REVIEW

File: Resources/views/elements/post-button.html.twig [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {% trans_default_domain 'EMSCoreBundle' %}{% if message is defined %}
 <div class="btn-group">
-  <button type="button" class="btn {% if class is defined %}{% else %}btn-sm{% endif %} btn-danger dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+  <button type="button" class="btn {% if class is defined %}{{ class }}{% else %}btn-sm btn-danger{% endif %} dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
```
Comment (dameert) [inline Resources/views/elements/post-button.html.twig]: To be verified with MDK, I think POST actions should always be btn-danger

Comment (ems-project) [reply Resources/views/elements/post-button.html.twig]: Agree, but it fix a bug

File: Repository/EnvironmentRepository.php [hunk 2]
```diff
@@ -22,6 +22,17 @@ public function findBy(array $criteria, array $orderBy = null, $limit = null, $o
...
     }
+    
+    /**
+     * @return array
+     */
+    public function findAllAliases()
+    {
+        $qb = $this->createQueryBuilder('e', 'e.alias');
+        $qb->select('e.alias, e.name, e.managed');
+
```
Comment (ems-project) [inline Repository/EnvironmentRepository.php]: I'm confuse, does the environment entity as an alias field or an aliases field

File: Resources/DoctrineMigrations/Version20180129102508.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
+<?php
```
Comment (ems-project) [inline Resources/DoctrineMigrations/Version20180129102508.php]: The migration script should be generated in the local-user-bundle (for mysql, pgsql and sqlite)

File: Resources/views/environment/index.html.twig [hunk 4]
```diff
@@ -75,7 +75,7 @@
 							<th>Action</th>
 						</tr>
 						{% for environment in environments %}
-							{{ 'Environmen\'s info'|debug({name: environment.name}) }}
+							{{ "Environmen's info"|debug({name: environment.name}) }}
```
Comment (ems-project) [inline Resources/views/environment/index.html.twig]: This is an interesting update. I must be from the past: https://stackoverflow.com/questions/3446216/what-is-the-difference-between-single-quoted-and-double-quoted-strings-in-php

File: Resources/views/environment/index.html.twig [hunk 5]
```diff
@@ -93,7 +93,7 @@
 						                </span>		
 									</td>
 									<td>{{ environment.alias }}</td>
-									<td>{{ environment.index }}</td>
```
Comment (ems-project) [inline Resources/views/environment/index.html.twig]: see other comments

File: Repository/EnvironmentRepository.php (hunk 2 above)
Comment (Davidmattei) [reply Repository/EnvironmentRepository.php]: Yes the environment entity has the following columns: id, name, alias, color, managed, ... The alias is the environment name prefixed with the instance_id parameter.

File: Entity/ManagedAlias.php [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+    
+    /**
+     * @var array
+     */
+    private $indexes = [];
+    
+    /**
+     * @var int
+     */
+    private $total;
```
[the remark this answers is not in the document]
Comment (Davidmattei) [reply Entity/ManagedAlias.php]: It's the total documents for the managed alias.
$this->client->count(['index' => 'aliasName'])

File: Resources/views/environment/index.html.twig (hunk 5 above)
Comment (Davidmattei) [reply Resources/views/environment/index.html.twig]: Fixed

File: Resources/views/elements/post-button.html.twig (hunk 1 above)
Comment (dameert) [reply Resources/views/elements/post-button.html.twig]: Ok, so no longer the red look obligation for Post buttons?

File: Resources/DoctrineMigrations/Version20180129102508.php (hunk 3 above)
Comment (Davidmattei) [reply Resources/DoctrineMigrations/Version20180129102508.php]: https://github.com/ems-project/EMSLocalUserBundle/pull/2/files

**Pull request merged**